server/implementation-cdi/src/main/java/io/smallrye/graphql/cdi/producer/SmallRyeContextAccessorProxy.java

```diff
@@ -0,0 +1,153 @@
+package io.smallrye.graphql.cdi.producer;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+
+import graphql.ExecutionInput;
+import graphql.schema.DataFetchingEnvironment;
+import io.smallrye.graphql.execution.context.SmallRyeContext;
+import io.smallrye.graphql.schema.model.Field;
+
+/**
+ * Class that serves as a proxy decorator for obtaining the current SmallRyeContext. It always calls
+ * SmallRyeContext.getContext() to obtain the current instance, and delegates to it.
+ * This is necessary because the SmallRyeContext is an immutable class, yet we, in some cases,
+ * need to be able to inject different instances of it during the serving of one HTTP request.
+ * This way, we make sure that an @Inject-ed SmallRyeContext is never cached and calls to it always check
+ * if there is a new context instance assigned to the current thread.
+ */
+@ApplicationScoped
```

Comment:
Should this not be `@RequestScope` ?

Reply:
Technically it does not matter what the scope is, this class does not hold any state, it only proxies a thread local variable.
For that reason, I believe application-scope is the best choice, because we avoid unnecessarily creating multiple instances.

Reply:
Ok cool ! Thanks